src/overlays/actors/ovl_Bg_Crace_Movebg/z_bg_crace_movebg.c

```diff
@@ -28,47 +44,256 @@ const ActorInit Bg_Crace_Movebg_InitVars = {
+        func_80A70A84(this, play);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A7099C.s")
+void func_80A70A84(BgCraceMovebg* this, PlayState* play) {
+    this->actionFunc = func_80A70A9C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A709E4.s")
+void func_80A70A9C(BgCraceMovebg* this, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A70A08.s")
+void BgCraceMovebg_Destroy(Actor* thisx, PlayState* play) {
+    BgCraceMovebg* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A70A84.s")
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    if (!(this->unk170 & 0x8)) {
+        switch (BGCRACEMOVEBG_GET_F(&this->dyna.actor)) {
+            case 0:
+            case 2:
+                Flags_UnsetSwitch(play, BGCRACEMOVEBG_GET_7F0(thisx));
+                Flags_UnsetSwitch(play, BGCRACEMOVEBG_GET_7F0(thisx) + 1);
+                break;
+            case 1:
+                Flags_UnsetSwitch(play, BGCRACEMOVEBG_GET_7F0(thisx));
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A70A9C.s")
+void BgCraceMovebg_Update(Actor* thisx, PlayState* play) {
+    BgCraceMovebg* this = THIS;
+    s32 pad;
+    Player* player = GET_PLAYER(play);
+    s16 tempRot;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/BgCraceMovebg_Destroy.s")
+    switch (BGCRACEMOVEBG_GET_F(&this->dyna.actor)) {
+        case 0:
+        case 2:
+            tempRot = this->dyna.actor.yawTowardsPlayer - this->dyna.actor.home.rot.y;
+            if ((tempRot >= -0x4000) && (tempRot <= 0x4000)) {
+                this->dyna.actor.shape.rot.y = this->dyna.actor.home.rot.y;
+            } else {
+                this->dyna.actor.shape.rot.y = this->dyna.actor.home.rot.y + 0x8000;
+            }
+        default:
+            this->actionFunc(this, play);
+            Math_Vec3f_Copy(&this->unk188, &player->bodyPartsPos[0]);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/BgCraceMovebg_Update.s")
+void func_80A70C04(BgCraceMovebg* this, PlayState* play) {
+    s32 pad;
+    Player* player = GET_PLAYER(play);
+    Vec3f intersect;
+    Vec3f diff;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A70C04.s")
+    if ((BGCRACEMOVEBG_GET_F(&this->dyna.actor) != 2) &&
+        (SubS_LineSegVsPlane(&this->dyna.actor.home.pos, &this->dyna.actor.home.rot, D_80A710AC, &this->unk188,
+                             &player->bodyPartsPos[0], &intersect)) &&
+        (Matrix_RotateYS(-this->dyna.actor.home.rot.y, MTXMODE_NEW),
+         Math_Vec3f_Diff(&player->bodyPartsPos[0], &this->dyna.actor.home.pos, &diff),
+         Matrix_MultVec3f(&diff, &this->unk178), (fabsf(this->unk178.x) < 100.0f)) &&
+        (this->unk178.y >= -10.0f) && (this->unk178.y <= 180.0f)) {
```

Comment:
```suggestion
    if ((BGCRACEMOVEBG_GET_F(&this->dyna.actor) != 2) &&
        SubS_LineSegVsPlane(&this->dyna.actor.home.pos, &this->dyna.actor.home.rot, D_80A710AC, &this->unk188,
                             &player->bodyPartsPos[0], &intersect)) {
        Matrix_RotateYS(-this->dyna.actor.home.rot.y, MTXMODE_NEW);
         Math_Vec3f_Diff(&player->bodyPartsPos[0], &this->dyna.actor.home.pos, &diff);
         Matrix_MultVec3f(&diff, &this->unk178);
         if ((fabsf(this->unk178.x) < 100.0f) && (this->unk178.y >= -10.0f) && (this->unk178.y <= 180.0f)) {
```
I feel like this should be able to be split, especially since the Matrix functions don't return anything and so shouldn't be a par t of the if